lexos/models/stats_model.py

```diff
@@ -0,0 +1,179 @@
+            if file_sizes[count] > median + 1.5 * iqr:
+                file_anomaly_iqr.update({label: 'large'})
+            elif file_sizes[count] < median - 1.5 * iqr:
+                file_anomaly_iqr.update({label: 'small'})
+
+        return CorpusInfo(q1=q1,
+                          q3=q3,
+                          iqr=iqr,
+                          median=median,
+                          average=average_file_size,
+                          num_file=num_file,
+                          file_sizes=list(file_sizes),
+                          file_names=labels,
+                          anomaly_iqr=file_anomaly_iqr,
+                          std_deviation=std_dev_file_size,
+                          anomaly_std_err=file_anomaly_std_err)
+
+    @staticmethod
+    def _get_file_info(count_list: np.ndarray, file_name: str) -> FileInfo:
+        """Gives statistics of a particular file in a given file list.
+
+        :param count_list: a list contains words count of a particular file.
+        :param file_name: the file name of that file.
+        """
+        assert np.sum(count_list) > 0, EMPTY_LIST_MESSAGE
+        # initialize remove all zeros from count_list
+        nonzero_count_list = count_list[count_list != 0]
+        num_word = np.size(nonzero_count_list)
+        total_word_count = int(sum(nonzero_count_list).item())
+        # 1 standard error analysis
+        average_word_count = round(total_word_count / num_word, 3)
+        # calculate the standard deviation
+        std_word_count = np.std(nonzero_count_list).item()
+
+        # 2 iqr analysis
+        median = np.median(nonzero_count_list).item()
+        q1 = np.percentile(nonzero_count_list, 25, interpolation="midpoint")
+        q3 = np.percentile(nonzero_count_list, 75, interpolation="midpoint")
+        iqr = q3 - q1
+        hapax = ((count_list == 1).sum()).item()
+
+        return FileInfo(q1=q1,
+                        q3=q3,
+                        iqr=iqr,
+                        hapax=hapax,
+                        median=median,
+                        average=average_word_count,
+                        num_word=num_word,
+                        file_name=file_name,
+                        word_count=nonzero_count_list,
+                        std_deviation=std_word_count,
+                        total_word_count=total_word_count)
+
+    def get_file_result(self) -> List[FileInfo]:
+        """Find statistics of all files and put each result into a list."""
+
+        file_info_list = \
+            [self._get_file_info(
+                count_list=self._doc_term_matrix.loc[[file_id]].values,
+                file_name=temp_label)
+             for file_id, temp_label in self._id_temp_label_map.items()]
+        return file_info_list
+
+    def get_corpus_result(self) -> CorpusInfo:
+        """Return stats for the whole corpus."""
+
+        return self._get_corpus_info()
+
+    @staticmethod
+    def get_token_type() -> str:
```

Comment:
I am kind of not sure how do I feel about this function. 

This function has to be impure whereas it really don't provide that much functionality...